src/components/select/select.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+
+interface Props extends React.DetailedHTMLProps<React.SelectHTMLAttributes<HTMLSelectElement>, HTMLSelectElement> {
+  options: Array<string>
+}
+
+const Select: React.FC<Props> = (props: Props) => {
```

Comment:
Você não precisa tipar o retorno do componente, typescript possui inferência de tipos, e sempre use suas props desestruturadas já
```suggestion
const Select = ({options, ...props}: Props) => {
```

Reply:
1 - No caso do button então eu poderia simplesmente fazer o mesmo que você fez pro Select e ficaria da forma abaixo? Ou existe uma forma ainda melhor do que essa pra esses cenários?
```
interface Props {
  variant: ButtonType
  label: string
}

const Button = ({variant, label, ...props}: Props) => {
```

2 - Supondo que meu Props possua muitos atributos, a melhor forma seria essa abaixo?
```
interface Props {
  variant: ButtonType
  label: string
  id: number
  lorem: string
}

const Button = (props: Props) => {
```

3 - Após fazer o ajuste pra deixar a interface dessa forma acima, passei a ter um erro (no onClick) no meu teste, quando chamo o meu componente Button, aparentemente por que o **onClick** não é esperado na minha Props:
```
render(<Button variant={ButtonType.Success} label={faker.lorem.word()} onClick={actionClicked} />)
```
Erro:
![image](https://github.com/raphael-avelar/form-react/assets/40543767/738319b9-83dc-490b-8a92-4671cb72ae6b)

Uma das soluções que encontrei, porém não sei se isso é uma boa prática, mas foi a seguinte, declarar o **onClick** dessa forma na interface Props:
```
interface Props {
  variant: ButtonType
  label: string
  onClick: () => void
}
```

4 - Utilizar a interface dessa forma acima (sem extender o **React.DetailedHTMLProps**) aparentemente te obriga a declarar todos os props que vão ser passados pro componente. Ele não entende os props padrão do componente.

Exemplo, eu estou trabalhando com o Input, se eu não usar o props extends "InputHTMLAttributes", eu preciso declarar até o próprio onChange (que de certa forma é nativo do componente, assim como o exemplo do button acima). Isso é correto?

Reply:
1 - sim
2 - usa desestruturado const Button = ({...}: Props) => {
3 - vc tem que passar o onClick pro seu componente como props. A interface define quais são obrigatorias e quais são opcionais, é sua responsabilidade cuidar disso
4 - Você pode deixar extendendo, não tem problema, a diferença é que pode existir atributos html que você não quer, e mesmo assim eles poderiam ser passados. Imagine que na sua aplicação você não aceita que os inputs sejam `disabled`. Se vc fizer a interface do seu wrapper extendendo os atributos padrão do input, ele vai ter o disabled acessivel.